src/android/MsgBox.java

```diff
@@ -18,109 +18,108 @@
  * Created by LuoWen on 2016/1/20.
  */
 public class MsgBox {
-    public static final String TAG = "MsgBox";
-    private Context mContext;
-    private MsgHelper msgHelper;
-
-    private Dialog noticeDialog;
-    private AlertDialog downloadDialog;
-    private ProgressBar downloadDialogProgress;
-    private Dialog errorDialog;
-
-    public MsgBox(Context mContext) {
-        this.mContext = mContext;
-        this.msgHelper = new MsgHelper(mContext.getPackageName(), mContext.getResources());
+  public static final String TAG = "MsgBox";
+  private Context mContext;
+  private MsgHelper msgHelper;
+
+  private Dialog noticeDialog;
+  private AlertDialog downloadDialog;
+  private ProgressBar downloadDialogProgress;
+  private Dialog errorDialog;
+
+  public MsgBox(Context mContext) {
+    this.mContext = mContext;
+    this.msgHelper = new MsgHelper(mContext.getPackageName(), mContext.getResources());
+  }
+
+  /**
+   * 显示软件更新对话框
+   *
+   * @param onClickListener
+   */
+  public Dialog showNoticeDialog(OnClickListener onClickListener) {
+    if (noticeDialog == null) {
+      LOG.d(TAG, "showNoticeDialog");
+      // 构造对话框
+      AlertDialog.Builder builder = new AlertDialog.Builder(mContext);
+      builder.setTitle(msgHelper.getString(MsgHelper.UPDATE_TITLE));
+      builder.setMessage(msgHelper.getString(MsgHelper.UPDATE_MESSAGE));
+      // 更新
+      builder.setPositiveButton(msgHelper.getString(MsgHelper.UPDATE_UPDATE_BTN), onClickListener);
+      noticeDialog = builder.create();
     }
 
-    /**
-     * 显示软件更新对话框
-     *
-     * @param onClickListener
-     */
-    public Dialog showNoticeDialog(OnClickListener onClickListener) {
-        if (noticeDialog == null) {
-            LOG.d(TAG, "showNoticeDialog");
-            // 构造对话框
-            AlertDialog.Builder builder = new AlertDialog.Builder(mContext);
-            builder.setTitle(msgHelper.getString(MsgHelper.UPDATE_TITLE));
-            builder.setMessage(msgHelper.getString(MsgHelper.UPDATE_MESSAGE));
-            // 更新
-            builder.setPositiveButton(msgHelper.getString(MsgHelper.UPDATE_UPDATE_BTN), onClickListener);
-            noticeDialog = builder.create();
-        }
-
-        if (!noticeDialog.isShowing()) noticeDialog.show();
-
-        noticeDialog.setCanceledOnTouchOutside(false);// 设置点击屏幕Dialog不消失
-        return noticeDialog;
+    if (!noticeDialog.isShowing())
+      noticeDialog.show();
+
+    noticeDialog.setCanceledOnTouchOutside(false);// 设置点击屏幕Dialog不消失
+    return noticeDialog;
+  }
+
+  /**
+   * 显示软件下载对话框
+   */
+  public Map<String, Object> showDownloadDialog(OnClickListener onClickListenerNeg, OnClickListener onClickListenerPos,
+      OnClickListener onClickListenerNeu, boolean showDialog) {
```

Comment:
Maybe split this up a bit more. This is really difficult to read. The original was easier.

- [ ] Done
- [ ] Approved